### PR TITLE
Add a test for agg std

### LINKED ISSUE #18734: ddof for np.std in df.agg changes depending on how given & lambda expression does not work correctly in a list of functions
#### Code Sample, a copy-pastable example if possible

```python
In [31]: import numpy as np

In [32]: import pandas as pd

In [33]: df = pd.DataFrame(np.arange(6).reshape(3, 2), columns=['A', 'B'])

In [34]: df
Out[34]:
   A  B
0  0  1
1  2  3
2  4  5

In [35]: df.agg(np.std)  # Behavior of ddof=0
Out[35]:
A    1.632993
B    1.632993
dtype: float64

In [36]: df.agg([np.std])  # Behavior of ddof=1
Out[36]:
       A    B
std  2.0  2.0

In [37]: # So how to get the ddof=0 behavior when giving a list of functions?

In [39]: df.agg([lambda x: np.std(x)])  # This gives a numerically unexpected result.
Out[39]:
         A        B
  <lambda> <lambda>
0      0.0      0.0
1      0.0      0.0
2      0.0      0.0

In [40]: df.agg([np.mean, lambda x: np.std(x)])  # This gives an error.
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-40-52f4ec4195b5> in <module>()
----> 1 df.agg([np.mean, lambda x: np.std(x)])

/Users/ikeda/.pyenv/versions/anaconda3-4.4.0/lib/python3.6/site-packages/pandas/core/frame.py in aggregate(self, func, axis, *args, **kwargs)
   4740         if axis == 0:
   4741             try:
-> 4742                 result, how = self._aggregate(func, axis=0, *args, **kwargs)
   4743             except TypeError:
   4744                 pass

/Users/ikeda/.pyenv/versions/anaconda3-4.4.0/lib/python3.6/site-packages/pandas/core/base.py in _aggregate(self, arg, *args, **kwargs)
    537             return self._aggregate_multiple_funcs(arg,
    538                                                   _level=_level,
--> 539                                                   _axis=_axis), None
    540         else:
    541             result = None

/Users/ikeda/.pyenv/versions/anaconda3-4.4.0/lib/python3.6/site-packages/pandas/core/base.py in _aggregate_multiple_funcs(self, arg, _level, _axis)
    594         # if we are empty
    595         if not len(results):
--> 596             raise ValueError("no results")
    597
    598         try:

ValueError: no results

```
#### Problem description

When using, e.g., `df.agg`, the `ddof` (degrees of freedom) value for the function `np.std` changes depending on how the function is given (single function or a list of functions), which may be so confusing for many people. I believe the behavior should be unified in some way.

Furthermore, I could not find the way to obtain to the `np.std` result with `ddof=0` by supplying it as one of the members of a list of functions. The `lambda` expression does not work well in a list of functions (this gives numerically unexpected results or even gives errors). This prohibits us to use many useful methods like `df.agg`, `df.apply`, and `df.describe` when we hope the `ddof=0` behavior. 

From https://github.com/pandas-dev/pandas/issues/13344, I guess Developers prefer the `ddof=1` behavior in pandas. So the expected behavior should be as below.

#### Expected Output
```
In [35]: df.agg(np.std)  # Behavior of ddof=1
Out[35]:
A    2.0
B    2.0
dtype: float64

In [38]: df.agg([lambda x: np.std(x)])  # To obtain the ddof=0 results
Out[38]:
                     A             B
<lambda>      1.632993      1.632993

In [41]: df.agg([np.mean, lambda x: np.std(x)])
                     A             B
mean          2.0           3.0
<lambda>      1.632993      1.632993
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None

pandas: 0.21.0
pytest: 3.0.7
pip: 9.0.1
setuptools: 27.2.0
Cython: 0.25.2
numpy: 1.13.3
scipy: 0.19.0
pyarrow: None
xarray: None
IPython: 5.3.0
sphinx: 1.5.6
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.3
blosc: None
bottleneck: 1.2.1
tables: 3.3.0
numexpr: 2.6.2
feather: None
matplotlib: 2.0.2
openpyxl: 2.4.7
xlrd: 1.0.0
xlwt: 1.2.0
xlsxwriter: 0.9.6
lxml: 3.7.3
bs4: 4.6.0
html5lib: 0.999
sqlalchemy: 1.1.9
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by steliospetrakis02**
- [X] closes #18734 
- [X] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [X] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

I believe that this test ensures the expected output. Let me know if you have any questions or concerns.

### CODE REVIEW

File: pandas/tests/groupby/aggregate/test_aggregate.py [hunk 1]
```diff
@@ -1577,3 +1577,9 @@ def test_agg_groupings_selection():
     )
     expected = DataFrame({"b": [6, 4], "c": [11, 7]}, index=index)
     tm.assert_frame_equal(result, expected)
+
+def test_agg_std():
+    df = pd.DataFrame({'A': [1, 2, 3], 'B': [4, 5, 6]})
```
Comment (mroeschke) [inline pandas/tests/groupby/aggregate/test_aggregate.py:1582]: Can you use the exact DataFrame from the test?

File: pandas/tests/groupby/aggregate/test_aggregate.py [hunk 2]
```diff
@@ -1577,3 +1577,9 @@ def test_agg_groupings_selection():
...
+    std = df.agg(np.std)
```
Comment (mroeschke) [inline pandas/tests/groupby/aggregate/test_aggregate.py:1583]:
```suggestion
    expected = df.agg(np.std)
```

Could you also compare `df.agg([np.std])`?

**Comment by mroeschke** [comment]
Can you move the test to `pandas/tests/apply/test_frame_apply.py`?

File: pandas/tests/groupby/aggregate/test_aggregate.py (hunk 2 above)
Comment (steliospetrakis02) [reply pandas/tests/groupby/aggregate/test_aggregate.py:1583]: Hello @mroeschke, I'm sorry but I'm not sure what you are asking. Do you want me to check the type of class? e.g (pandas.core.series.Series)

Comment (mroeschke) [reply pandas/tests/groupby/aggregate/test_aggregate.py:1583]: Just asking to add a test to also check the result of `df.agg([np.std])`  as well

**Comment by steliospetrakis02** [comment]
I believe that this will fulfill your request, but please let me know if there is anything else I can do. @mroeschke

File: pandas/tests/apply/test_frame_apply.py [hunk 3]
```diff
@@ -1665,3 +1665,13 @@ def foo2(x, b=2, c=0):
...
+
+def test_agg_std():
+    df = pd.DataFrame(np.arange(6).reshape(3, 2), columns=['A', 'B'])
+
+    result = df.agg(np.std)
+    expected = pd.Series({'A': 2.0, 'B': 2.0}, dtype=float)
+    tm.assert_series_equal(result, expected)    
+
+    result = df.agg([np.std])
+    assert isinstance(result, pd.DataFrame)
```
Comment (mroeschke) [inline pandas/tests/apply/test_frame_apply.py:1677]: Please construct the expected result and use `tm.assert_frame_equal`

**Comment by mroeschke** [comment]
Looks like a few of the code check builds are failing

**Comment by steliospetrakis02** [comment]
@mroeschke , I have fixed the code checks that were failing, and I have constructed the tests  as  your instructions.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Awesome! Thanks @steliospetrakis02

**Comment by steliospetrakis02** [comment]
> Awesome! Thanks @steliospetrakis02

You're welcome! I thank you for your guidance and assistance as I take my first steps into contributing to pandas

**Pull request merged**